content/articles/creating-a-pyspark-data-frame-a-beginners-guide/index.md

```diff
@@ -0,0 +1,159 @@
+|Division|Kiswahili|Biology|Agriculture|Mathematics|
++--------+---------+-------+-----------+-----------+
+|       C|       80|     70|         85|         91|
+|       B|       80|     70|         85|         91|
+|       A|       80|     78|         96|         92|
+|       A|       92|     70|         89|         96|
++--------+---------+-------+-----------+-----------+
+```
+> To see the whole code for this [click here](https://colab.research.google.com/drive/1Ykwifphdj2mLQ9rhKuX309aXq-tGurTD?usp=sharing)
+### Creating Python data frame from an external file using PySpark
+Our external Files will be imported using the spark session's `read()` function. Data frame object will be created as the result of this action. The techniques to import these file formats are practically comparable, and one may import them with minimal effort. In contrast to the previous method, this one is simple and requires a spark session to produce a PySpark data frame from a Resilient Distributed Dataset.
+
+#### 1. Importing the libraries:
+```Python
+from pyspark.sql import SparkSession
+```
+#### 2. Creating the spark session
+```python
+sparkSession = SparkSession.builder.appName('Creating PySpark dataframe from an external file').getOrCreate()
+```
+Using `appName(),` we can retrieve the application's name by passing in a string. It is possible to leverage an existing spark session or build a new one using the `getOrCreate()` method.
+#### 3. Read the external files into the PySpark data frame
+The external file format that we will use and import comprises a CSV file.
+
+```python
+ourCSV_file = sparkSession.read.csv('IncomeAndGender.csv', sep = ',', inferSchema = True, header = True)
+```
+Delimiters for the CSV file were given (which is a comma). Afterward, the inferSchema property was set to True. CSV files will be automatically transformed into PySpark data frames with this command. 
+
+Let's have a look at our PySpark data frame's data kinds now:
+```python
+print(type(ourCSV_file))
+```
+Output:
+```bash
+<class 'pyspark.sql.dataframe.DataFrame'>
+```
+We can also check the schema:
+```python
+ourCSV_file.printSchema()
+```
+Output:
+```bash
+root
+ |-- Salary: double (nullable = true)
+ |-- Gender: integer (nullable = true)
+ |-- AgeNumber: double (nullable = true)
+ |-- PhD: string (nullable = true)
+```
+> To see the complete code for this [click here](https://colab.research.google.com/drive/1u04j9muZ1FdDvr2oxeuSGST5Lva9es9j?usp=sharing). In addition, the CSV file that we used here was initially created in our notebook.
+
+### Other methods 
+#### 1. Converting PySpark data frame to a Pandas data frame
+The PySpark data frame can be converted to a Pandas data frame, and Pandas techniques can be used on our data frame, which is incredibly convenient.
+```python
+csvToPandas = ourCSV_file.toPandas()
+```
+To ensure that everything we're doing is correct, let's look at the data type of our new data frame.

```

Comment:
To ensure that everything works correctly, let's look at the data type of our new data frame.